src/FormNetworks/quadraticformnetwork.jl

```diff
@@ -0,0 +1,75 @@
+default_index_map = prime
+default_inv_index_map = noprime
+
+struct QuadraticFormNetwork{V,FormNetwork<:BilinearFormNetwork{V},IndexMap} <:
+       AbstractFormNetwork{V}
+  formnetwork::FormNetwork
+  dual_index_map::IndexMap
+end
+
+bilinear_formnetwork(qf::QuadraticFormNetwork) = qf.formnetwork
+function QuadraticFormNetwork(
+  operator::AbstractITensorNetwork,
+  bra::AbstractITensorNetwork,
+  ket::AbstractITensorNetwork;
+  dual_index_map=default_index_map,
+  kwargs...,
+)
+  return QuadraticFormNetwork(
+    BilinearFormNetwork(operator, bra, ket; kwargs...), dual_index_map
+  )
+end
+
+#Needed for implementation, forward from bilinear form
+for f in [
+  :operator_vertex_suffix,
+  :bra_vertex_suffix,
+  :ket_vertex_suffix,
+  :tensornetwork,
+  :data_graph,
+  :data_graph_type,
+]
+  @eval begin
+    function $f(qf::QuadraticFormNetwork, args...; kwargs...)
+      return $f(bilinear_formnetwork(qf), args...; kwargs...)
+    end
+  end
+end
+
+dual_index_map(qf::QuadraticFormNetwork) = qf.dual_index_map
+function copy(qf::QuadraticFormNetwork)
+  return QuadraticFormNetwork(copy(bilinear_formnetwork(qf)), dual_index_map(qf))
+end
+
+function QuadraticFormNetwork(
+  operator::AbstractITensorNetwork,
+  ket::AbstractITensorNetwork;
+  dual_index_map=default_index_map,
+  kwargs...,
+)
+  bra = map_inds(dual_index_map, dag(ket))
+  return QuadraticFormNetwork(operator, bra, ket; dual_index_map, kwargs...)
+end
+
+function QuadraticFormNetwork(
+  ket::AbstractITensorNetwork; dual_index_map=default_index_map, kwargs...
+)
+  s = siteinds(ket)
+  operator_inds = union_all_inds(s, dual_index_map(s; links=[]))
+  operator = delta_network(operator_inds)
+  return QuadraticFormNetwork(operator, ket; kwargs...)
+end
```

Comment:
I think it would be simpler if both of these just directly call the corresponding `BilinearFormNetwork` constructor and then wrap that in a `QuadraticFormNetwork`.